### DISCUSSION TITLE
Roaming and MOBIKE

### DISCUSSION THREAD

**Discussion by ww20** [strongswan/strongswan#1052] [Q&A]
Hello,

With regard to roaming, is that tied to mobike being enabled? We've explicitly disabled mobike in our config (see below). In our case, the source IP for an existing SA disappears and a new path is sought. Is there more configuration required other than setting mobike=no? We'd rather these SA's are deleted (if possible) by DPD. We can post more if necessary, apologies ahead of time for that if needed, given the posting requirements. Have to prune some information before posting. Our strongswan version is 5.7.2.

cyc_strongswan[30600]: 07[KNL] fddf:9df:c86a:0:201:4499:7a1:6a4d disappeared from eth0
...
cyc_strongswan[30600]: 01[KNL] creating acquire job for policy fddf:9df:c86a:0:201:44b9:974c:5515/128[tcp/52984] === fddf:9df:c86a:0:201:4464:912:f21e/128[tcp/5433] with reqid {3}
cyc_strongswan[30600]: 07[CFG] ignoring acquire, connection attempt pending
cyc_strongswan[30600]: 09[KNL] creating acquire job for policy fddf:9df:c86a:0:201:4499:7a1:6a4d/128[tcp/43496] === fddf:9df:c86a:0:201:44f1:aa27:961d/128[tcp/5433] with reqid {3}
cyc_strongswan[30600]: 09[IKE] initiating IKE_SA host-host-v6[114] to fddf:9df:c86a:0:201:44f1:aa27:961d
cyc_strongswan[30600]: 08[IKE] old path is not available anymore, try to find another
cyc_strongswan[30600]: 08[IKE] looking for a route to fddf:9df:c86a:0:201:44f1:aa27:961d ...
cyc_strongswan[30600]: 08[IKE] reauthenticating IKE_SA due to address change
cyc_strongswan[30600]: 08[IKE] reinitiating IKE_SA host-host-v6[114]
cyc_strongswan[30600]: 08[IKE] initiating IKE_SA host-host-v6[114] to fddf:9df:c86a:0:201:44f1:aa27:961d

connections {
    host-host-v6 { 
        local_addrs= %any6
        remote_addrs = %any6
        local {
            auth = pubkey
            certs = ipsec.cert.pem
            id = "C=XX, ST=WW, O=YY, CN=ZZ"
        }
        remote {
            auth = pubkey
        }
        children {
            cluster-icm-v6 {
                local_ts = fd13:31f3:2b27::201:443a:85c6:147c/64
                remote_ts = fd13:31f3:2b27::201:443a:85c6:147c/64
                esp_proposals = aesxxxxxxxxxxxxx
                start_action = start
                mode = transport
            }
       } 
        version = 2
        mobike = no
        proposals = aesxxxxxxxxxxxxxxxxxxxxxxxxxxx
        dpd_delay = 15 
   }
}

**Comment by tobiasbrunner** [comment]
> With regard to roaming, is that tied to mobike being enabled?

No, it just won't use MOBIKE. After disabling MOBIKE, you have to explicitly configure a local IP address (`local_addrs`) to ignore roaming events. If you can't do that, you could disable roaming events globally via `charon.plugins.kernel-netlink.roam_events` in [strongswan.conf](https://docs.strongswan.org/docs/5.9/config/strongswanConf.html#_charon_plugins_kernel_netlink).

**Reply by ww20** [reply]
Thank you. Setting roam_events to no seemed to work. I had a follow up question, does setting roam_events to no have any implications for NAT-T? It's possible we may end up using NAT-T in the future.Thanks again.

**Reply by tobiasbrunner** [reply]
No, that has nothing to do with NAT-T.
